Test test_pid_check_samepid_two_processes fails on FreeBSD

If I understand this error correctly, the code did not raise a `PidFileAlreadyRunningError` when it was supposed to. I do not yet know why.

I have no problem running the test suite on Linux (Python 3.7.1), only on FreeBSD (Python 3.6.6).

```
======================================================================
FAIL: test_pid.test_pid_check_samepid_two_processes
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/ports/devel/py-pid/work-py36/pid-2.2.1/tests/test_pid.py", line 280, in test_pid_check_samepid_two_processes
    pidfile_proc2.create()
  File "/usr/local/lib/python3.6/contextlib.py", line 88, in __exit__
    next(self.gen)
  File "/usr/ports/devel/py-pid/work-py36/pid-2.2.1/tests/test_pid.py", line 42, in raising
    raise AssertionError("Failed to throw exception of type(s) %s." % (", ".join(exc_type.__name__ for exc_type in exc_types),))
AssertionError: Failed to throw exception of type(s) PidFileAlreadyRunningError, PidFileAlreadyLockedError.
-------------------- >> begin captured logging << --------------------
PidFile: DEBUG: <pid.PidFile object at 0x807eee2c8> entering setup
PidFile: DEBUG: <pid.PidFile object at 0x807eee2c8> create pidfile: /tmp/setup.py.pid
PidFile: DEBUG: <pid.PidFile object at 0x807eee2c8> check pidfile: /tmp/setup.py.pid
PidFile: DEBUG: <pid.PidFile object at 0x807eee368> entering setup
PidFile: DEBUG: <pid.PidFile object at 0x807eee368> create pidfile: /tmp/setup.py.pid
PidFile: DEBUG: <pid.PidFile object at 0x807eee368> check pidfile: /tmp/setup.py.pid
PidFile: DEBUG: <pid.PidFile object at 0x807eee2c8> closing pidfile: /tmp/setup.py.pid
PidFile: DEBUG: <pid.PidFile object at 0x807eee368> closing pidfile: /tmp/setup.py.pid
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 32 tests in 0.030s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=32 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=32 errors=0 failures=1>
```

Also, on [line 279](https://github.com/trbs/pid/blob/0efff53af4554dfc8e06a2627809baa84d13b732/tests/test_pid.py#L279), I don't understand why it is expecting `PidFileAlreadyLockedError`? If I remove it, so it only expects `PidFileAlreadyRunningError`, the test still passes (on Linux). So, what exactly is supposed be happening in this test case?

--- a/tests/test_pid.py
+++ b/tests/test_pid.py
@@ -266,18 +266,22 @@ def test_pid_check_samepid():
         pidfile.close()
 
 
-def test_pid_raises_already_running_when_samepid_and_two_different_pids():
+@patch("os.getpid")
+@patch("os.kill")
+def test_pid_raises_already_running_when_samepid_and_two_different_pids(
+    mock_getpid, mock_kill,
+):
     pidfile_proc1 = pid.PidFile()
     pidfile_proc2 = pid.PidFile(allow_samepid=True)
 
     try:
-        with patch('pid.os.getpid') as mgetpid:
-            mgetpid.return_value = 1
-            pidfile_proc1.create()
+        mock_getpid.return_value = 1
+        pidfile_proc2.create()
 
-            mgetpid.return_value = 2
-            with raising(pid.PidFileAlreadyRunningError):
-                pidfile_proc2.create()
+        mock_getpid.return_value = 2
+        with raising(pid.PidFileAlreadyRunningError):
+            pidfile_proc2.create()
+
     finally:
         pidfile_proc1.close()
         pidfile_proc2.close()

--- a/tests/test_pid.py
+++ b/tests/test_pid.py
@@ -276,7 +276,7 @@ def test_pid_check_samepid_two_processes():
             pidfile_proc1.create()
 
             mgetpid.return_value = 2
-            with raising(pid.PidFileAlreadyRunningError, pid.PidFileAlreadyLockedError):
+            with raising(pid.PidFileAlreadyRunningError):
                 pidfile_proc2.create()
     finally:
         pidfile_proc1.close()

--- a/tests/test_pid.py
+++ b/tests/test_pid.py
@@ -266,7 +266,7 @@ def test_pid_check_samepid():
         pidfile.close()
 
 
-def test_pid_check_samepid_two_processes():
+def test_pid_raises_already_running_when_samepid_and_two_different_pids():
     pidfile_proc1 = pid.PidFile()
     pidfile_proc2 = pid.PidFile(allow_samepid=True)
 

